test/jit/test_await.py

```diff
@@ -0,0 +1,331 @@
+
+
+class TestAwait(JitTestCase):
+    def test_await_python(self):
+        def foo(x: int) -> int:
+            return x + 13
+        aw: Await[int] = torch.jit._awaitable(foo, 13)
+        self.assertTrue(aw.fn()(*aw.args()) == torch.jit._awaitable_wait(aw))
+        nw = torch.jit._awaitable_nowait(33)
+        self.assertTrue(nw.is_nowait())
+        self.assertTrue(nw.args() == (33,))
+
+    def test_await_type_python(self):
+        def foo() -> Tensor:
+            return torch.randn()
+        awaits = torch.jit.annotate(List[Await[Tensor]], [])
+        awaits.append(torch.jit._awaitable(foo))
+
+    def test_script(self):
+        def delayed(z: int) -> int:
+            return z + 3
+
+        @torch.jit.script
+        def fn(x: Tensor):
+            aw: Await[int] = torch.jit._awaitable(delayed, 99)
+            a = torch.eye(2)
+            b = torch.jit._awaitable_wait(aw)
+            return a + b + x
+
+        inp = torch.zeros(2)
+
+        sm = torch.jit.script(fn)
+        out = fn(inp)
+        script_out = sm(inp)
+        self.assertTrue(torch.allclose(script_out, out))
+
+    def test_nowait(self):
+        @torch.jit.script
+        def fn(x: Tensor):
+            aw = torch.jit._awaitable_nowait(13)
+            a = torch.eye(2)
+            b = torch.jit._awaitable_wait(aw)
+            return a + b + x
+
+        inp = torch.zeros(2)
+
+        sm = torch.jit.script(fn)
+        out = fn(inp)
+        script_out = sm(inp)
+        self.assertTrue(torch.allclose(script_out, out))
+
+    def test_nowait_class(self):
+        class C(object):
+            def __init__(self, a: Tensor, b: Tensor):
+                self._a = a
+                self._b = b
+
+            def a(self) -> Tensor:
+                return self._a
+
+        @torch.jit.script
+        def fn(x: Tensor):
+            aw = torch.jit._awaitable_nowait(C(torch.zeros(2), torch.ones(2)))
+            _a = torch.eye(2)
+            c = torch.jit._awaitable_wait(aw)
+            return _a + c.a() + x
+
+        make_global(C)
+        inp = torch.zeros(2)
+
+        sm = torch.jit.script(fn)
+        out = fn(inp)
+        script_out = sm(inp)
+        self.assertTrue(torch.allclose(script_out, out))
+
+
+    def test_await_class_arg(self):
+
+        @torch.jit.script
+        class C(object):
+            def __init__(self, a: Tensor, b: Tensor):
+                self.__a = a
+                self.__b = b
+
+            def a(self) -> Tensor:
+                return self.__a
+
+        make_global(C)
+
+        @torch.jit.script
+        def delayed(c: C) -> Tensor:
+            return c.a()
+
+        @torch.jit.script
+        def fn(x: Tensor):
+            c = C(torch.zeros(2), torch.ones(2))
+            aw = torch.jit._awaitable(delayed, c)
+            _a = torch.eye(2)
+            c2_t = torch.jit._awaitable_wait(aw)
+            return _a + c2_t + x
+        inp = torch.zeros(2)
+
+        sm = torch.jit.script(fn)
```

Comment:
this is not doing anything, since `def fn` is already marked with `@torch.jit.script`. i.e. sm and fn should behave identically because they are both scripted.

also, for these can you add explicit checks for expected behavior instead of trying to compare vs. eager? in many of these cases it's not clear to me what the behavior is expected to be, so it might be nicer to explicitly show in the test what the expected output is.

(although, of course it is useful to have some eager and eager vs. scripted tests)

(same for many of the other tests in this file)